### ISSUE TITLE
Covariance matrices must be positive definite. Add regularization to avoid this error

### ISSUE THREAD

**Issue by loic001** [alexandrebarachant/pyRiemann#65]
Hi,

Over all, thanks for your lib !

In this simple example
```
from pyriemann.estimation import Covariances
from pyriemann.utils.mean import mean_riemann
cov = Covariances().fit_transform(X_train)
mean_riemann(cov)
```

I get an error

`Covariance matrices must be positive definite. Add regularization to avoid this error.`

X_train is correct format with shape (trials, channels, n_samples)

Maybe I'm sailing in the wrong direction but I investigated a bit and I got NaN values after the logm function at line 61 in mean_riemann function (utils/mean.py)

 `J += sample_weight[index] * logm(tmp) `

 The first iterations for gradient descent until k=3 are going well, but NaN values appear at the third iteration.

Thanks for your help !

**Comment by alexandrebarachant** [comment]
hey,
this happens usually when the covariance are not positive definite. There is mainly two reason for that : 
- you have more channel than time sample 
- your data is rank-deficient, i.e. you applied common average reference or some other methods (max_filter for MEG data ?)

if you can not fix the root cause of the problem, then add regularization :
`cov = Covariances('oas').fit_transform(X_train)`
and you will be just fine.

Note that in most cases, adding regularization does not impact negatively the performances

**Comment by loic001** [comment]
Thank you for your quick and helpful response !
